src/MySqlConnector/Core/ResultSet.cs

```diff
@@ -147,12 +147,12 @@ public async Task ReadResultSetHeaderAsync(IOBehavior ioBehavior)
 						// server supports metadata caching, but has re-sent it, so something has changed since last prepare/execution
 						if (Session.SupportsCachedPreparedMetadata && DataReader.LastUsedPreparedStatement is { } preparedStatement)
 							preparedStatement.Columns = ColumnDefinitions;
+					}
 
-						if (!Session.SupportsDeprecateEof)
-						{
-							payload = await Session.ReceiveReplyAsync(ioBehavior, CancellationToken.None).ConfigureAwait(false);
-							EofPayload.Create(payload.Span);
-						}
+					if (!Session.SupportsDeprecateEof)
+					{
+						payload = await Session.ReceiveReplyAsync(ioBehavior, CancellationToken.None).ConfigureAwait(false);
+						EofPayload.Create(payload.Span);
```

Comment:
Is this change correct/intentional?

It seems like it could require an additional EOF packet for the case when metadata from a prepared statement is reused. If so, it feels like integration tests shouldn't pass both before and after this change. Perhaps `MARIADB_CLIENT_CACHE_METADATA` always implies `CLIENT_DEPRECATE_EOF` so that this condition is always `false` and the `if` block is never entered?

Reply:
Just reviewed https://mariadb.com/kb/en/result-set-packets/ and confirmed that this change is correct.

Reply:
yes, this might have been done through another PR. This is a correction that might normally never occurs (there is no version that support skipping metadata without supporting EOF) but better to correct that just in case.